src/main/java/com/example/timefit/domain/room/service/RoomService.java

```diff
@@ -1,5 +1,118 @@
 package com.example.timefit.domain.room.service;
 
+import com.example.timefit.domain.room.dto.RoomCreateRequest;
+import com.example.timefit.domain.room.dto.RoomDeleteMessage;
+import com.example.timefit.domain.room.dto.RoomResponse;
+import com.example.timefit.domain.room.dto.RoomUpdateRequest;
+import com.example.timefit.domain.room.entity.Room;
+import com.example.timefit.domain.room.entity.RoomDate;
+import com.example.timefit.domain.room.repository.RoomRepository;
+import com.example.timefit.domain.user.entity.User;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
 public class RoomService {
-  
+
+    private final RoomRepository roomRepository;
+
+    @Transactional
+    public RoomResponse createRoom(RoomCreateRequest dto, User owner) {
+        String newInviteCode = UUID.randomUUID().toString();
+
+        LocalTime startTime = parseTime(dto.getStartTime());
+        LocalTime endTime = parseTime(dto.getEndTime());
+
+        Room room = new Room(
+                dto.getTitle(),
+                newInviteCode,
+                owner,
+                startTime,
+                endTime
+        );
+        room.onPrePersist();
+
+        for (LocalDate date : dto.getDates()) {
+            room.getDates().add(new RoomDate(date, room));
+        }
+
+        Room savedRoom = roomRepository.save(room);
+
+        return new RoomResponse(savedRoom);
+    }
+
+    @Transactional(readOnly = true)
+    public List<RoomResponse> getAllRooms() {
+        return roomRepository.findAll().stream()
+                .map(RoomResponse::new)
+                .collect(Collectors.toList());
```

Comment:
`stream()`을 사용하셨네요! 반복문이 아니라 `stream`을 사용한 이유가 있으신가요?

```suggestion
        return roomRepository.findAll().stream()
                .map(RoomResponse::new)
                .toList();
```

Reply:
RoomResponse에서 따로 함수로 만들어서 사용해야 하는 번거로움을 없애기 위해 스트림을 사용했던 것이 기억나서 stream을 사용했는데 찾아보니 for문에 효율이 더 좋은 것 같아 교체 해보겠습니다.

Reply:
stream을 사용하지 말라는 의도의 리뷰는 아니였습니다.
다만 반복문이 익숙하고 stream API의 장점을 제대로 인지하지 못하고 있는데 어떤 흐름으로 stream이 나왔나 궁금했습니다

Reply:
제가 공부하기로는 병렬처리로 인해 최적화와 효율성이 좋다고 알고 있습니다.
근데 찾아보니 속도면에서는 또 for문이 좋다고 하는데 어떤 것을 쓰는게 좋을까요? 둘 다 상관 없을 것 같기도 한데 혹시 몰라 고민입니다!

Reply:
저는 stream api를 선호하는 편입니다. 성능상 차이는 사실상 신경 안써도 될만한 차이이고, stream api를 썻을때 훨씬 직관적이고 읽기 쉬운 코드가 된다고 생각합니다.

다만 리뷰를 남겼던 의도는 "stream api를 어떻게 사용하게 되었나?" 입니다. 분명 반복문이 익숙할테고, stream api를 사용해야겠다 라고 생각하고 사용했다면 그 이유가 당연히 있어야 된다고 생각합니다.

Reply:
아하 넵 알겠습니다. 저도 stream을 사용을 해보고 싶어서 stream으로 다시 변경하겠습니다.